### ISSUE TITLE
Not working on my T9 AIVI PLUS

### ISSUE THREAD

**Issue by huongminh** [nicoduj/homebridge-deebotEcovacs#71]
I installed the plugin successfully and can see all the switches and fan. On the log I could see my Ecovacs T9 AIVI PLus is found at discovery with the correct ID (E0541xxxxxx). However, when I tried to play with the switches and fan in my Home app, the T9 would not respond at all except for the Bip switch. When I turned this sound stateless switch, I would hear a sound from my T9 so I guess the communication between the home app and the T9 should be OK, as well as my configuration. Please update the plugin so it works with the latest model of Ecovacs Deebot T9. Thank you

**Comment by nicoduj** [comment]
Hi,

please post full logs in debug mode . Don't know if this model is fully supported by the lib, have to check.

**Comment by huongminh** [comment]
Thank you for your quick response. I got this in the journalctl log of my homebridge instance
[DeebotEcovacs] INFO - stopping deebots discovery, number of deebots found : 1
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Discovered Deebot : E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Edge Cleaning for E05414875C09HHXD0292 : true
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Spot Cleaning for E05414875C09HHXD0292 : true
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - SpotArea Cleaning for E05414875C09HHXD0292 : false
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - CustomArea Cleaning for E05414875C09HHXD0292 : false
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Battery Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Fan Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Main Switch Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Sound stateless Switch Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Motion Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Auto stateless Switch Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Edge stateless Switch Service for E05414875C09HHXD0292
Jun 09 00:01:17 raspberrypi homebridge[15142]: [09/06/2021, 00:01:17] [DeebotEcovacs] INFO - Creating Spot stateless Switch Service for E05414875C09HHXD0292

how can I run in debug mode? thank you

**Comment by nicoduj** [comment]
see homebridge doc and activate debug mode : DEBUG=*

**Comment by huongminh** [comment]
This is what I get when I turn on the "Fan" service

Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.453Z HAP-NodeJS:EventedHTTPServer:Connection [::ffff:192.168.1.75] HTTP request: /characteristics
Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.453Z HAP-NodeJS:HAPServer [CD:22:3D:E4:CE:36] HAP Request: PUT /characteristics
Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.454Z HAP-NodeJS:Accessory [Homebridge-roomba AB63] Processing characteristic set: {"characteristics":[{"aid":12,"iid":15,"value":1}]}
Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.462Z HAP-NodeJS:EventedHTTPServer [::ffff:192.168.1.75] Muting event '12.15' notification for this connection since it originated here.
Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.463Z HAP-NodeJS:Accessory [Homebridge-roomba AB63] Setting Characteristic "On" to value 1
Jun 09 01:17:47 raspberrypi homebridge[2215]: 2021-06-08T18:17:47.465Z HAP-NodeJS:EventedHTTPServer:Connection [::ffff:192.168.1.75] HTTP Response is finished
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.075Z mqttjs:client _checkPing :: checking ping...
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.075Z mqttjs:client _checkPing :: ping response received. Clearing flag and sending `pingreq`
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.075Z mqttjs:client _sendPacket :: (fffmzhk0d83f8bb5@ecouser/5f033c97) ::  start
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.075Z mqttjs:client sendPacket :: packet: { cmd: 'pingreq' }
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.076Z mqttjs:client sendPacket :: emitting `packetsend`
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.076Z mqttjs:client sendPacket :: writing to stream
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.076Z mqtt-packet:writeToStream generate called
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.076Z mqtt-packet:writeToStream generate: packet.cmd: pingreq
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.076Z mqttjs:client sendPacket :: writeToStream result true
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.327Z mqttjs:client writable stream :: parsing buffer
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.328Z mqtt-packet:parser parse: current state: _parseHeader
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.329Z mqtt-packet:parser _parseHeader: packet: Packet { cmd: 'pingresp', retain: false, qos: 0, dup: false, length: -1, topic: null, payload: null }
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.329Z mqtt-packet:parser parse: state complete. _stateCounter is now: 1
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.329Z mqtt-packet:parser parse: packet.length: -1, buffer list length: 1
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.329Z mqtt-packet:parser _parseVarByteNum
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.330Z mqtt-packet:parser _parseVarByteNum: result: { bytes: 1, value: 0 }
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.330Z mqtt-packet:parser _parseLength 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.330Z mqtt-packet:parser parse: state complete. _stateCounter is now: 2
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.331Z mqtt-packet:parser parse: packet.length: 0, buffer list length: 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.334Z mqtt-packet:parser _parsePayload: payload BufferListStream {
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _bufs: [],
Jun 09 01:17:48 raspberrypi homebridge[2215]:   length: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _readableState: ReadableState {
Jun 09 01:17:48 raspberrypi homebridge[2215]:     objectMode: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     highWaterMark: 16384,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     buffer: BufferList { head: null, tail: null, length: 0 },
Jun 09 01:17:48 raspberrypi homebridge[2215]:     length: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     pipes: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     pipesCount: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     flowing: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     ended: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     endEmitted: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     reading: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     sync: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     needReadable: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     emittedReadable: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     readableListening: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     resumeScheduled: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     paused: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     emitClose: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     autoDestroy: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     destroyed: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     defaultEncoding: 'utf8',
Jun 09 01:17:48 raspberrypi homebridge[2215]:     awaitDrain: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     readingMore: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     decoder: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     encoding: null
Jun 09 01:17:48 raspberrypi homebridge[2215]:   },
Jun 09 01:17:48 raspberrypi homebridge[2215]:   readable: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _events: [Object: null prototype] {},
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _eventsCount: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _maxListeners: undefined,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   _writableState: WritableState {
Jun 09 01:17:48 raspberrypi homebridge[2215]:     objectMode: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     highWaterMark: 16384,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     finalCalled: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     needDrain: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     ending: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     ended: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     finished: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     destroyed: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     decodeStrings: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     defaultEncoding: 'utf8',
Jun 09 01:17:48 raspberrypi homebridge[2215]:     length: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     writing: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     corked: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     sync: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     bufferProcessing: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     onwrite: [Function (anonymous)],
Jun 09 01:17:48 raspberrypi homebridge[2215]:     writecb: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     writelen: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     bufferedRequest: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     lastBufferedRequest: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     pendingcb: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     prefinished: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     errorEmitted: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     emitClose: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     autoDestroy: false,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     bufferedRequestCount: 0,
Jun 09 01:17:48 raspberrypi homebridge[2215]:     corkedRequestsFree: CorkedRequest {
Jun 09 01:17:48 raspberrypi homebridge[2215]:       next: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:       entry: null,
Jun 09 01:17:48 raspberrypi homebridge[2215]:       finish: [Function (anonymous)]
Jun 09 01:17:48 raspberrypi homebridge[2215]:     }
Jun 09 01:17:48 raspberrypi homebridge[2215]:   },
Jun 09 01:17:48 raspberrypi homebridge[2215]:   writable: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   allowHalfOpen: true,
Jun 09 01:17:48 raspberrypi homebridge[2215]:   [Symbol(kCapture)]: false
Jun 09 01:17:48 raspberrypi homebridge[2215]: }
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.334Z mqtt-packet:parser _parsePayload complete result: true
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.335Z mqtt-packet:parser parse: state complete. _stateCounter is now: 3
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.335Z mqtt-packet:parser parse: packet.length: 0, buffer list length: 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.335Z mqtt-packet:parser _newPacket
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.336Z mqtt-packet:parser _newPacket: parser emit packet: packet.cmd: pingresp, packet.payload: null, packet.length: 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.336Z mqttjs:client parser :: on packet push to packets array.
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.337Z mqtt-packet:parser _newPacket: new packet
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.337Z mqtt-packet:parser parse: state complete. _stateCounter is now: 4
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.337Z mqtt-packet:parser parse: packet.length: -1, buffer list length: 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.337Z mqtt-packet:parser parse: exited while loop. packet: -1, buffer list length: 0
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.338Z mqttjs:client work :: getting next packet in queue
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.338Z mqttjs:client work :: packet pulled from queue
Jun 09 01:17:48 raspberrypi homebridge[2215]: 2021-06-08T18:17:48.338Z mqttjs:client _handlePacket :: emitting packetreceive

**Comment by huongminh** [comment]
Let me clarify that my T9 setup is not in the same local wifi network with my Pi3 on which I installed the this  plugin (I put my T9 in office while my Pi is at home). Anyway, the bip switch just works fine. The rest are not working, including the battery service.

**Comment by mrbungle64** [comment]
@huongminh 

I only need the follwing line with the `class` value from the normal log:

```
[15.5.2021, 18:37:02] [DeebotEcovacs] INFO - getDeebots : [{...,"class":"b2jqs4",...}}]
```

Should be something like `b2jqs4`

**Comment by mrbungle64** [comment]
> Let me clarify that my T9 setup is not in the same local wifi network with my Pi3 on which I installed the this plugin (I put my T9 in office while my Pi is at home). Anyway, the bip switch just works fine. The rest are not working, including the battery service.

The Deebot does not necessarily have to be in the same local wifi network.

**Comment by huongminh** [comment]
Sorry I only find these:
[09/06/2021, 01:26:25] [DeebotEcovacs] INFO - stopping deebots discovery, number of deebots found : 1
Jun 09 01:26:25 raspberrypi homebridge[6383]: [09/06/2021, 01:26:25] [DeebotEcovacs] INFO - Discovered Deebot : E05414875C09HHXD0292
Jun 09 01:26:25 raspberrypi homebridge[6383]: [09/06/2021, 01:26:25] [DeebotEcovacs] INFO - Edge Cleaning for E05414875C09HHXD0292 : true
Jun 09 01:26:25 raspberrypi homebridge[6383]: [09/06/2021, 01:26:25] [DeebotEcovacs] INFO - Spot Cleaning for E05414875C09HHXD0292 : true
Jun 09 01:26:25 raspberrypi homebridge[6383]: [09/06/2021, 01:26:25] [DeebotEcovacs] INFO - SpotArea Cleaning for E05414875C09HHXD0292 : false
Jun 09 01:26:25 raspberrypi homebridge[6383]: [09/06/2021, 01:26:25] [DeebotEcovacs] INFO - CustomArea Cleaning for E05414875C09HHXD0292 : false

**Comment by mrbungle64** [comment]
@huongminh 

Please check the line above this one:
```
[09/06/2021, 01:26:25] [DeebotEcovacs] INFO - stopping deebots discovery, number of deebots found : 1
```

**Comment by huongminh** [comment]
Here you go, I post some line above and under:

Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.394Z mqtt-packet:writeToStream writeNumberCached: <Buffer 00 04>
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.394Z mqtt-packet:writeToStream writeString: MQTT
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.395Z mqtt-packet:writeToStream writeNumberCached: number: 60
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.395Z mqtt-packet:writeToStream writeNumberCached: <Buffer 00 3c>
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.395Z mqtt-packet:writeToStream writeNumberCached: number: 33
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.396Z mqtt-packet:writeToStream writeNumberCached: <Buffer 00 21>
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.396Z mqtt-packet:writeToStream writeString: fffmzhk0d83f8bb5@ecouser/5f033c97
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.397Z mqtt-packet:writeToStream writeNumberCached: number: 24
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.397Z mqtt-packet:writeToStream writeNumberCached: <Buffer 00 18>
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.397Z mqtt-packet:writeToStream writeString: fffmzhk0d83f8bb5@ecouser
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.398Z mqtt-packet:writeToStream writeNumberCached: number: 198
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.398Z mqtt-packet:writeToStream writeNumberCached: <Buffer 00 c6>
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.398Z mqtt-packet:writeToStream writeString: V_2_eyJjIjoiNjA3OTA2OTExMmUyZTlhMmFkOWMxZDMyIiwidSI6ImZmZm16aGswZDgzZjhiYjUiLCJyIjoiNWYwMzNjOTciLCJ0IjoiYSIsImlhdCI6MTYyMzE3NzM1NSwiZXhwIjoxNjIzNzgyMTU1fQ.CNHJS1oLFKoadfpdTNY2jLwL41pjDvCBq7p9tgUHimg
Jun 09 01:35:57 raspberrypi homebridge[9140]: 2021-06-08T18:35:57.399Z mqttjs:client sendPacket :: writeToStream result true
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - stopping deebots discovery, number of deebots found : 1
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - Discovered Deebot : E05414875C09HHXD0292
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - Edge Cleaning for E05414875C09HHXD0292 : true
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - Spot Cleaning for E05414875C09HHXD0292 : true
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - SpotArea Cleaning for E05414875C09HHXD0292 : false
Jun 09 01:35:57 raspberrypi homebridge[9140]: [09/06/2021, 01:35:57] [DeebotEcovacs] INFO - CustomArea Cleaning for E05414875C09HHXD0292 : false
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.490Z mqttjs:client writable stream :: parsing buffer
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.492Z mqtt-packet:parser parse: current state: _parseHeader
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.500Z mqtt-packet:parser _parseHeader: packet: Packet { cmd: 'connack', retain: false, qos: 0, dup: false, length: -1, topic: null, payload: null }
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.501Z mqtt-packet:parser parse: state complete. _stateCounter is now: 1
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.502Z mqtt-packet:parser parse: packet.length: -1, buffer list length: 0
Jun 09 01:35:58 raspberrypi homebridge[9140]: 2021-06-08T18:35:58.502Z mqtt-packet:parser parse: exited while loop. packet: -1, buffer list length: 0
Jun 09 01:35:58 raspberrypi h

**Comment by mrbungle64** [comment]
@huongminh 

You have to enable debug Mode in Homebridge settings first:

- Startup Options
   - Homebridge Debug Mode -D

**Comment by mrbungle64** [comment]
@nicoduj 

Why there's `INFO` in the output of debug log level?

```
[15.5.2021, 18:37:02] [DeebotEcovacs] INFO - getDeebots : [{...,"class":"b2jqs4",...}}]
```

**Comment by huongminh** [comment]
I did here as per your request and I could see more logs as I posted:

pi@raspberrypi:/etc/default $ cat homebridge-roomba
# Defaults / Configuration options for homebridge
# The following settings tells homebridge where to find the config.json file and where to persist the data (i.e. pairing and others)
HOMEBRIDGE_OPTS= -U /var/homebridge-roomba -P /var/homebridge-roomba/node_modules

# If you uncomment the following line, homebridge will log more 
# You can display this via systemd's journalctl: journalctl -f -u homebridge
DEBUG=*

**Comment by huongminh** [comment]
I uncommented the DEBUG=*

**Comment by mrbungle64** [comment]
@huongminh

> I did here as per your request and I could see more logs as I posted:
>
> pi@raspberrypi:/etc/default $ cat homebridge-roomba

This was not my request 😉

**Comment by huongminh** [comment]
Sorry. As I understand I find the Debug=* in that config file for my homebridge service. Pls tell me how I should do the get more logs

**Comment by mrbungle64** [comment]
@huongminh 

> Sorry. As I understand I find the Debug=* in that config file for my homebridge service. Pls tell me how I should do the get more logs

https://github.com/nicoduj/homebridge-deebotEcovacs/issues/71#issuecomment-857003193

You can check the Homebridge log in your Homebridge UI

**Comment by nicoduj** [comment]
I , can't check but try this option in config : showInfoLogs to true

**Comment by huongminh** [comment]
Finally I got it. Thank you for your patience. I had to put a -D in the homebridge execution

[09/06/2021, 01:53:44] [DeebotEcovacs] INFO - connected
Jun 09 01:53:45 raspberrypi homebridge[14317]: [09/06/2021, 01:53:45] [DeebotEcovacs] INFO - getDeebots : [{"did":"cb1059e7","name":"E05414292","class":"kw9ayx","resource":"vJ7R","nick":null,"company":"eco-ng","bindTs":1622907128356,"service":{"jmq":"jmq-ngiot-cn.dc.cn.ecouser.net","mqs":"api-ngiot.dc-cn.cn.ecouser.net"}}]

**Comment by huongminh** [comment]
by the way if I set a name  for my robot in config.json then the robot is not found any more in discovery. So I had to get the default name of E05414875C09HHXD0292

**Comment by mrbungle64** [comment]
@huongminh @nicoduj 

I added the new class to the config.
It'll be available with the next version of the library.

**Comment by huongminh** [comment]
Thank you very much! you did a great job! when can I get the updated version? and does it fix the "name" issue?

**Comment by nicoduj** [comment]
I will need to update the plugin, will try to do ASAP . Concerning the name it is the nickName I use so you have to rename it in deebot app I think

**Comment by mrbungle64** [comment]
@nicoduj 

I'll let you know when I published a new version on npm

**Comment by huongminh** [comment]
thank you very much!

**Comment by mrbungle64** [comment]
@nicoduj 

I just published 0.6.1-alpha.13 of the library.

**Comment by nicoduj** [comment]
Hi, will update the lib today with latest 0.6.5
